tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -347,6 +349,59 @@ async def test_form_unsupported_device(
     assert result3["type"] is FlowResultType.CREATE_ENTRY
     assert result3["title"] == f"Powerview Generation {api_version}"
     assert result3["data"] == {CONF_HOST: "1.2.3.4", CONF_API_VERSION: api_version}
-    assert result3["result"].unique_id == "A1B2C3D4E5G6H7"
+    assert result3["result"].unique_id == MOCK_SERIAL
 
     assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.usefixtures("mock_hunterdouglas_hub")
+@pytest.mark.parametrize("api_version", [1, 2, 3])
+async def test_migrate_entry(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    api_version: int,
+) -> None:
+    """Test migrate to newest version."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"host": "1.2.3.4"},
+        unique_id=MOCK_SERIAL,
+        version=1,
+        minor_version=1,
+    )
+
+    assert entry.version == 1
+    assert entry.minor_version == 1
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.version == 1
+    assert entry.minor_version == 2
+
+    # Add entries with int unique_id
+    entity_registry.async_get_or_create(
+        domain="cover",
+        platform="hunterdouglas_powerview",
+        unique_id=123,
+        config_entry=entry,
+    )
+    # Add entries with a str unique_id not starting with entry.unique_id
+    entity_registry.async_get_or_create(
+        domain="cover",
+        platform="hunterdouglas_powerview",
+        unique_id="old_unique_id",
+        config_entry=entry,
+    )
```

Comment:
Move these before the `async_setup`. Home Assistant should migrate itself we should not have to manually call `_migrate_entities`

Reply:
done 👍